mmdet/visualization/wandb_visualizer.py

```diff
@@ -0,0 +1,313 @@
+            out_file (str): Path to output file. Defaults to None.
+                Invalid.
+            pred_score_thr (float): The threshold to visualize the bboxes
+                and masks. Defaults to 0.3.
+            step (int): Global step value to record. Defaults to 0.
+        """
+        if self.save_dir is None:
+            warnings.warn('`WandbVisBackend` is not initialized '
+                          'because save_dir is None. Return directly.')
+            return
+        if show is True:
+            warnings.warn('Wandb Visualizer does not have a local window '
+                          'display function, ignore the show parameter.')
+        if out_file is not None:
+            warnings.warn('Wandb Visualizer does not have the function '
+                          'of saving files to the local disk, '
+                          'ignore the out_file parameter.')
+
+        if self.class_id_to_label is None:
+            assert 'CLASSES' in self.dataset_meta
+            self.class_id_to_label = {
+                id + 1: name
+                for id, name in enumerate(self.dataset_meta['CLASSES'])
+            }
+            self.class_set = self._wandb.Classes([{
+                'id': id,
+                'name': name
+            } for id, name in self.class_id_to_label.items()])
+
+        if data_sample is not None:
+            data_sample = data_sample.numpy()
+
+        gt_img_data = None
+        pred_img_data = None
+
+        if draw_gt and data_sample is not None:
+            gt_img_data = image
+            if 'gt_instances' in data_sample:
+                gt_img_data = self._draw_instances(image,
+                                                   data_sample.gt_instances)
+            # TODO: Support panoptic_seg
+
+        if draw_pred and data_sample is not None:
+            pred_img_data = image
+            if 'pred_instances' in data_sample:
+                pred_instances = data_sample.pred_instances
+                pred_instances = pred_instances[
+                    pred_instances.scores > pred_score_thr]
+                pred_img_data = self._draw_instances(
+                    image, pred_instances, log_gt=False)
+            # TODO: Support panoptic_seg
+
+        if name in self._record_tables:
+            table = self._record_tables[name]
+        else:
+            if gt_img_data is not None and pred_img_data is not None:
+                table = self._wandb.Table(
+                    columns=['ground_truth', 'prediction'])
+            elif gt_img_data is not None:
+                table = self._wandb.Table(columns=['ground_truth'])
+            else:
+                table = self._wandb.Table(columns=['prediction'])
+            self._record_tables[name] = table
+
+        if gt_img_data is not None and pred_img_data is not None:
+            table.add_data(gt_img_data, pred_img_data)
+        elif gt_img_data is not None:
+            table.add_data(gt_img_data)
+        else:
```

Comment:
elif pred_img_data is not None:
     table.add_data(pred_img_data)
else:
     warnings.warn(' At least one of gt_img_data and pred_img_data is not None! ')